purchase_order_line_deep_sort/models/purchase_order.py

```diff
@@ -32,8 +32,12 @@ def _sort_purchase_line(self):
         def resolve_subfields(obj, line_order):
             subfields = line_order.split('.')
             res = obj
+            str_fields = (fields.Char, fields.Text)
             for subfield in subfields:
-                res = getattr(res, subfield)
+                if isinstance(res._fields[subfield], str_fields):
+                    res = getattr(res, subfield) or ''
+                else:
+                    res = getattr(res, subfield)
```

Comment:
IMHO, only one line should work,
res = getattr(res, subfield) or ''
We would prefer to have the test case or example where you need If condition (may be a good comment on code)

Reply:
Mhuuu!! But or '' os not always, if you order by a boolean field you do not want an empty string.
I will do tests for this use case... :smile:

Reply:
@JayVora-SerpentCS I change isinstance function by fields.column_type method.